src/cryptography/hazmat/backends/openssl/decode_asn1.py

```diff
@@ -710,6 +710,10 @@ def _asn1_string_to_utf8(backend, asn1_string):
     buf = backend._ffi.new("unsigned char **")
     res = backend._lib.ASN1_STRING_to_UTF8(buf, asn1_string)
     if res == -1:
+        str_bytes = _asn1_string_to_bytes(backend, asn1_string)
+        if asn1_string.type == int('0x12', 16):
```

Comment:
I don't like this approach to handling this type. I'd rather that we checked the string type before we ever called `_asn1_string_to_utf8` and handled switching there rather than trying it after an error is returned.